common/lib/xmodule/xmodule/partitions/tests/test_partitions.py

```diff
@@ -325,9 +325,19 @@ def test_get_user_group_id_for_partition(self):
 
         # switch to the second group and verify that it is returned for the user
         self.user_partition.scheme.current_group = groups[1]    # pylint: disable=no-member
-        group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #self.assertEqual(group2_id, groups[1].id)    # pylint: disable=no-member
+
+        partition_service2 = StaticPartitionService(
+            [self.user_partition],
+            user=Mock(username='ma', email='ma@edx.org', is_staff=False, is_active=True),
+            course_id=self.course.id,
+            track_function=Mock()
+        )
+        group2_id = partition_service2.get_user_group_id_for_partition(user_partition_id)
```

Comment:
So this is something I need to clean up and better message in the tests. The reason this is here is because of the cache on a `PartitionService`. If it doesn't get a cache object explicitly passed in, it will create one of its own, so it will last the lifetime of that object. In the actual runtime, we're passing in a request-cache so that every instance of PartitionService that's made (and there can be a lot of them) share that cached info -- but the cache dies at the end of the request.